routes/notifications.js

```diff
@@ -0,0 +1,161 @@
+const express = require("express");
+const router = express.Router();
+const Notification = require("../models/Notification");
+const joi = require("@hapi/joi");
+
+const notificationValidation = joi.object({
+  userId: joi.string().required(),
+  type: joi.string().required(),
+  isRead: joi.boolean(),
+  date: joi.date(),
+  content: joi.string().required().max(200),
+});
+
+const updateNotificationValidation = joi.object({
+  isRead: joi.boolean().required(),
+});
+
+// http://localhost/api/notifications ///////////////////////////////////////////////////////
+// GET
+// Returns all notifications
+
+router.get("/", async (req, res) => {
+  try {
+    const notifications = await Notification.find();
+    res.status(200).json(notifications);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      message: "Error code 500: Failed to process request",
+    });
+  }
+});
+
+router.get("/findById/:notificationId", async (req, res) => {
+  try {
+    const notifications = await Notification.find({
+      _id: req.params.notificationId,
+    });
+    res.status(200).json(notifications);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      message: "Error code 500: Failed to process request",
+    });
+  }
+});
+
+router.get("/findByUserId/:userId", async (req, res) => {
+  try {
+    const notifications = await Notification.find({
+      userId: req.params.userId,
+    });
+    res.status(200).json(notifications);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      message: "Error code 500: Failed to process request",
+    });
+  }
+});
+
+router.post("/", async (req, res) => {
+  const reqBody = req.body;
+
+  const { error } = notificationValidation.validate(reqBody);
+  if (error) {
+    return res.status(400).json({
+      message: error.details[0].message,
+    });
+  }
+
+  try {
+    const notification = new Notification({
+      userId: reqBody.userId,
+      type: reqBody.type,
+      content: reqBody.content,
+    });
+    const savedNotification = await notification.save();
+    res.status(200).json(savedNotification);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      message: "Error code 500: Failed to process request",
+    });
+  }
+});
+
+router.patch("/updateOneNotification/:notificationId", async (req, res) => {
```

Comment:
I can only envision the client modifying the "isRead" field, maybe simplify to suit only this purpose?

Reply:
By this I mean changing the path to perhaps /:notificationId/markAsRead

Reply:
okay yeah thats a good point, will change